c/bam_stats.c

```diff
@@ -32,13 +32,13 @@
 #include "khash.h"
 
 static char *input_file = NULL;
-static char *output_file;
-static char *ref_file;
+static char *output_file = NULL;
+static char *ref_file = NULL;
 static int rna = 0;
 int grps_size = 0;
 stats_rd_t*** grp_stats;
 static char *bas_header = "bam_filename\tsample\tplatform\tplatform_unit\tlibrary\treadgroup\tread_length_r1\tread_length_r2\t#_mapped_bases\t#_mapped_bases_r1\t#_mapped_bases_r2\t#_divergent_bases\t#_divergent_bases_r1\t#_divergent_bases_r2\t#_total_reads\t#_total_reads_r1\t#_total_reads_r2\t#_mapped_reads\t#_mapped_reads_r1\t#_mapped_reads_r2\t#_mapped_reads_properly_paired\t#_gc_bases_r1\t#_gc_bases_r2\tmean_insert_size\tinsert_size_sd\tmedian_insert_size\t#_duplicate_reads\n";
-static char *rg_line_pattern = "%s\t%s\t%s\t%s\t%s\t%s\t%ld\t%ld\t%lld\t%lld\t%lld\t%lld\t%lld\t%lld\t%lld\t%lld\t%lld\t%lld\t%lld\t%lld\t%lld\t%lld\t%lld\t%.3f\t%.3f\t%.3f\t%lld\n";
+static char *rg_line_pattern = "%s\t%s\t%s\t%s\t%s\t%s\t%d\t%d\t%d\t%d\t%d\t%d\t%d\t%d\t%d\t%d\t%d\t%d\t%d\t%d\t%d\t%d\t%d\t%.3f\t%.3f\t%.3f\t%d\n";
```

Comment:
There was. During my regression testing I noticed that some of the numbers (read_length_r2 in particular) was being calculated correctly, but printed incorrectly. Changing *rg_line_pattern solved the problem.

I'm really sorry, but I no longer have a note of the particular bam file I was using for testing, and I don't seem to be able to reproduce the problem with a random selection of bam files that I've just tried. I should have made sure I wrote some unit tests for this.

The other option I perhaps should have tried was the PRI macros, but I wanted to keep it simple.